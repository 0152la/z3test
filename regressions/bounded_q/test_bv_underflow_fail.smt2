(define-sort W () (_ BitVec 8))
(declare-fun qw (W W) Bool)


; fails since (bvsub y #x01) can be very large (underflow)
(assert (forall ((x W) (y W)) 
			(implies
				(and 
					(bvuge x #x00) (bvule x (bvsub y #x01))
					(bvuge y #x00) (bvule y #x04))
				(qw x y))
		))


(apply (then nnf expand_bounded_quantifiers))