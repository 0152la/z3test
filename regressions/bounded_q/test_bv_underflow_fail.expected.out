(goals
(goal
  (forall ((x (_ BitVec 8)) (y (_ BitVec 8)))
    (or (not (bvule y #x04)) (not (bvule x (bvadd #xff y))) (qw x y)))
  :precision precise :depth 5)
)
(goals
(goal
  (qw #x00 #x00)
  (qw #x01 #x00)
  (qw #x02 #x00)
  (qw #x03 #x00)
  (qw #x04 #x00)
  (qw #x05 #x00)
  (qw #x06 #x00)
  (qw #x07 #x00)
  (qw #x08 #x00)
  (qw #x09 #x00)
  (qw #x0a #x00)
  (qw #x0b #x00)
  (qw #x0c #x00)
  (qw #x0d #x00)
  (qw #x0e #x00)
  (qw #x0f #x00)
  (qw #x10 #x00)
  (qw #x11 #x00)
  (qw #x12 #x00)
  (qw #x13 #x00)
  (qw #x14 #x00)
  (qw #x15 #x00)
  (qw #x16 #x00)
  (qw #x17 #x00)
  (qw #x18 #x00)
  (qw #x19 #x00)
  (qw #x1a #x00)
  (qw #x1b #x00)
  (qw #x1c #x00)
  (qw #x1d #x00)
  (qw #x1e #x00)
  (qw #x1f #x00)
  (qw #x20 #x00)
  (qw #x21 #x00)
  (qw #x22 #x00)
  (qw #x23 #x00)
  (qw #x24 #x00)
  (qw #x25 #x00)
  (qw #x26 #x00)
  (qw #x27 #x00)
  (qw #x28 #x00)
  (qw #x29 #x00)
  (qw #x2a #x00)
  (qw #x2b #x00)
  (qw #x2c #x00)
  (qw #x2d #x00)
  (qw #x2e #x00)
  (qw #x2f #x00)
  (qw #x30 #x00)
  (qw #x31 #x00)
  (qw #x32 #x00)
  (qw #x33 #x00)
  (qw #x34 #x00)
  (qw #x35 #x00)
  (qw #x36 #x00)
  (qw #x37 #x00)
  (qw #x38 #x00)
  (qw #x39 #x00)
  (qw #x3a #x00)
  (qw #x3b #x00)
  (qw #x3c #x00)
  (qw #x3d #x00)
  (qw #x3e #x00)
  (qw #x3f #x00)
  (qw #x40 #x00)
  (qw #x41 #x00)
  (qw #x42 #x00)
  (qw #x43 #x00)
  (qw #x44 #x00)
  (qw #x45 #x00)
  (qw #x46 #x00)
  (qw #x47 #x00)
  (qw #x48 #x00)
  (qw #x49 #x00)
  (qw #x4a #x00)
  (qw #x4b #x00)
  (qw #x4c #x00)
  (qw #x4d #x00)
  (qw #x4e #x00)
  (qw #x4f #x00)
  (qw #x50 #x00)
  (qw #x51 #x00)
  (qw #x52 #x00)
  (qw #x53 #x00)
  (qw #x54 #x00)
  (qw #x55 #x00)
  (qw #x56 #x00)
  (qw #x57 #x00)
  (qw #x58 #x00)
  (qw #x59 #x00)
  (qw #x5a #x00)
  (qw #x5b #x00)
  (qw #x5c #x00)
  (qw #x5d #x00)
  (qw #x5e #x00)
  (qw #x5f #x00)
  (qw #x60 #x00)
  (qw #x61 #x00)
  (qw #x62 #x00)
  (qw #x63 #x00)
  (qw #x64 #x00)
  (qw #x65 #x00)
  (qw #x66 #x00)
  (qw #x67 #x00)
  (qw #x68 #x00)
  (qw #x69 #x00)
  (qw #x6a #x00)
  (qw #x6b #x00)
  (qw #x6c #x00)
  (qw #x6d #x00)
  (qw #x6e #x00)
  (qw #x6f #x00)
  (qw #x70 #x00)
  (qw #x71 #x00)
  (qw #x72 #x00)
  (qw #x73 #x00)
  (qw #x74 #x00)
  (qw #x75 #x00)
  (qw #x76 #x00)
  (qw #x77 #x00)
  (qw #x78 #x00)
  (qw #x79 #x00)
  (qw #x7a #x00)
  (qw #x7b #x00)
  (qw #x7c #x00)
  (qw #x7d #x00)
  (qw #x7e #x00)
  (qw #x7f #x00)
  (qw #x80 #x00)
  (qw #x81 #x00)
  (qw #x82 #x00)
  (qw #x83 #x00)
  (qw #x84 #x00)
  (qw #x85 #x00)
  (qw #x86 #x00)
  (qw #x87 #x00)
  (qw #x88 #x00)
  (qw #x89 #x00)
  (qw #x8a #x00)
  (qw #x8b #x00)
  (qw #x8c #x00)
  (qw #x8d #x00)
  (qw #x8e #x00)
  (qw #x8f #x00)
  (qw #x90 #x00)
  (qw #x91 #x00)
  (qw #x92 #x00)
  (qw #x93 #x00)
  (qw #x94 #x00)
  (qw #x95 #x00)
  (qw #x96 #x00)
  (qw #x97 #x00)
  (qw #x98 #x00)
  (qw #x99 #x00)
  (qw #x9a #x00)
  (qw #x9b #x00)
  (qw #x9c #x00)
  (qw #x9d #x00)
  (qw #x9e #x00)
  (qw #x9f #x00)
  (qw #xa0 #x00)
  (qw #xa1 #x00)
  (qw #xa2 #x00)
  (qw #xa3 #x00)
  (qw #xa4 #x00)
  (qw #xa5 #x00)
  (qw #xa6 #x00)
  (qw #xa7 #x00)
  (qw #xa8 #x00)
  (qw #xa9 #x00)
  (qw #xaa #x00)
  (qw #xab #x00)
  (qw #xac #x00)
  (qw #xad #x00)
  (qw #xae #x00)
  (qw #xaf #x00)
  (qw #xb0 #x00)
  (qw #xb1 #x00)
  (qw #xb2 #x00)
  (qw #xb3 #x00)
  (qw #xb4 #x00)
  (qw #xb5 #x00)
  (qw #xb6 #x00)
  (qw #xb7 #x00)
  (qw #xb8 #x00)
  (qw #xb9 #x00)
  (qw #xba #x00)
  (qw #xbb #x00)
  (qw #xbc #x00)
  (qw #xbd #x00)
  (qw #xbe #x00)
  (qw #xbf #x00)
  (qw #xc0 #x00)
  (qw #xc1 #x00)
  (qw #xc2 #x00)
  (qw #xc3 #x00)
  (qw #xc4 #x00)
  (qw #xc5 #x00)
  (qw #xc6 #x00)
  (qw #xc7 #x00)
  (qw #xc8 #x00)
  (qw #xc9 #x00)
  (qw #xca #x00)
  (qw #xcb #x00)
  (qw #xcc #x00)
  (qw #xcd #x00)
  (qw #xce #x00)
  (qw #xcf #x00)
  (qw #xd0 #x00)
  (qw #xd1 #x00)
  (qw #xd2 #x00)
  (qw #xd3 #x00)
  (qw #xd4 #x00)
  (qw #xd5 #x00)
  (qw #xd6 #x00)
  (qw #xd7 #x00)
  (qw #xd8 #x00)
  (qw #xd9 #x00)
  (qw #xda #x00)
  (qw #xdb #x00)
  (qw #xdc #x00)
  (qw #xdd #x00)
  (qw #xde #x00)
  (qw #xdf #x00)
  (qw #xe0 #x00)
  (qw #xe1 #x00)
  (qw #xe2 #x00)
  (qw #xe3 #x00)
  (qw #xe4 #x00)
  (qw #xe5 #x00)
  (qw #xe6 #x00)
  (qw #xe7 #x00)
  (qw #xe8 #x00)
  (qw #xe9 #x00)
  (qw #xea #x00)
  (qw #xeb #x00)
  (qw #xec #x00)
  (qw #xed #x00)
  (qw #xee #x00)
  (qw #xef #x00)
  (qw #xf0 #x00)
  (qw #xf1 #x00)
  (qw #xf2 #x00)
  (qw #xf3 #x00)
  (qw #xf4 #x00)
  (qw #xf5 #x00)
  (qw #xf6 #x00)
  (qw #xf7 #x00)
  (qw #xf8 #x00)
  (qw #xf9 #x00)
  (qw #xfa #x00)
  (qw #xfb #x00)
  (qw #xfc #x00)
  (qw #xfd #x00)
  (qw #xfe #x00)
  (qw #xff #x00)
  (qw #x00 #x01)
  (qw #x00 #x02)
  (qw #x01 #x02)
  (qw #x00 #x03)
  (qw #x01 #x03)
  (qw #x02 #x03)
  (qw #x00 #x04)
  (qw #x01 #x04)
  (qw #x02 #x04)
  (qw #x03 #x04)
  :precision precise :depth 5)
)
